src/en/zeroscans/src/eu/kanade/tachiyomi/extension/en/zeroscans/ZeroScans.kt

```
@@ -0,0 +1,326 @@
+
+    class SortFilter(sorts: List<ZeroScans.Sort>) :
+        Filter.Sort("Sort by", sorts.map { it.name }.toTypedArray(), Selection(3, false))
+
+    class Sort(val name: String, val type: String)
+
+    private val sortList = listOf(
+        Sort("Alphabetic", "alphabetic"),
+        Sort("Rating", "rating"),
+        Sort("Chapter Count", "chapter_count"),
+        Sort("Bookmark Count", "bookmark_count"),
+        Sort("View Count", "view_count")
+    )
+
+    class RankingsHeader :
+        Filter.Header("Note: Genre, Sort, Status filter and Search query")
+    class RankingsHeader2 :
+        Filter.Header("are not applied to rankings")
+
+    class RankingsFilter(rankings: List<Ranking>) :
+        Filter.Sort("Rankings", rankings.map { it.name }.toTypedArray(), Selection(0, false))
+
+    class Ranking(val name: String, val type: String? = null)
+
+    private val rankingList = listOf(
+        Ranking("None"),
+        Ranking("All Time", "all-time"),
+        Ranking("Weekly", "weekly"),
+        Ranking("Monthly", "monthly")
+    )
+
+    // Helpers
+    private fun comicsDataRequest(): Request {
+        return GET("$baseUrl/$API_PATH/comics")
+    }
+
+    private fun comicsDataParse(response: Response): ZeroScansComicsDataDto {
+        return json.decodeFromString<ZeroScansResponseDto<ZeroScansComicsDataDto>>(response.body!!.string()).data
+    }
+
+    private fun updateComicsData() {
+        val response = client.newCall(comicsDataRequest()).execute()
+        comicsDataParse(response).let {
+            genreList = it.genres.map { genreDto ->
+                Genre(genreDto.name, genreDto.id)
+            }
+            statusList = it.statuses.map { statusDto ->
+                Status(statusDto.name, statusDto.id)
+            }
+            comicList = it.comics
+            rankings = it.rankings
+        }
+    }
+
+    // Unused Stuff
+    override fun imageUrlParse(response: Response): String = throw Exception("Not Used")
```

Maybe `throw UnsupportedOperationException("Not used.")` to make these explicit and conform with the contributing guidelines

---

Alternatively, could just return `""` too in this case. In the other ones, better to follow Vetle suggestion.